packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -182,10 +184,34 @@ export const render = async (req, res, next) => {
         req,
         res,
         location,
-        config
+        config,
+        routerContext: {},
+        effectContext: {requests: []}
     }
     try {
-        renderResult = renderApp(args)
+        // Pre-render app to get useProps requests
+        await renderApp(args)
+
+        // Call all the useProps functions
+        const effectPromises = args.effectContext.requests.map((request) => request.fireEffect())
+        const hooksProps = await Promise.all(effectPromises)
+
+        // Turn array into a map.
+        const hooksPropsMap = hooksProps.reduce((acc, curr) => {
+            const [key, value] = Object.entries(curr)[0]
+            return {
+                ...acc,
+                [key]: value
+            }
+        }, {})
+
+        // Set the args with the new updated app state.
+        args.appState = {
+            ...appState,
+            hookProps: hooksPropsMap
+        }
```

Comment:
It's really, really hard to see here that `hookProps` is of the form `{"uuid": [PROPS_HERE]}`. I had to read and re-read, but I think I understand. I'm a little spooked by relying on `useUid` to match the saved props back up to the `useProps` call, but if it works and `useUid` provides that guarantee, why not!

On this line you seem to be depending on the order of `entries`. I can't see why and it doesn't seem like a good idea to me:

```
const [key, value] = Object.entries(curr)[0]
```

Do you know that `curr` is an object that only has one key in it or something?

Reply:
`const [key, value] = Object.entries(curr)[0]` is the result of me googling how to turn a key value object into a tuple. This code is ugly, would be cleaned up if we go forward with it. I appreciate pointing it out. 

> It's really, really hard to see here that hookProps is of the form {"uuid": [PROPS_HERE]}

Yup. It would be nice to clean that up too. I have some idea's. We could be really basic and require a key for those hooks to be used and better abstract that data resolving so it's easier to understand.